pimd/pim_bsm.c

```diff
@@ -1056,13 +1043,22 @@ static uint32_t hash_calc_on_grp_rp(struct prefix group, struct in_addr rp,
 	/* in_addr stores ip in big endian, hence network byte order
 	 * convert to uint32 before processing hash
 	 */
+#if PIM_IPV == 4
 	grpaddr = ntohl(group.u.prefix4.s_addr);
+#else
+	grpaddr = *(group.u.prefix6.s6_addr32);
```

Comment:
This doesn't look like it calculates the hash correctly? It only uses the first 32 bits of an IPv6 address.